Contest_API.md

```diff
@@ -1604,24 +1603,24 @@ Judgements for submissions in the contest.
 
 The following endpoints are associated with judgements:
 
-| Endpoint                         | Mime-type        | Required? | Description
-| :------------------------------- | :--------------- | :-------- | :----------
-| `/contests/<id>/judgements`      | application/json | yes       | JSON array of all judgements with properties as defined in the table below.
-| `/contests/<id>/judgements/<id>` | application/json | yes       | JSON object representing a single judgement with properties as defined in the table below.
+| Endpoint                         | Mime-type        | Description
+| :------------------------------- | :--------------- | :----------
+| `/contests/<id>/judgements`      | application/json | JSON array of all judgements with properties as defined in the table below.
+| `/contests/<id>/judgements/<id>` | application/json | JSON object representing a single judgement with properties as defined in the table below.
 
 Properties of judgement objects:
 
-| Name                 | Type    | Required? | Nullable? | Description
-| :------------------- | :------ | :-------- | :-------- | :----------
-| id                   | ID      | yes       | no        | Identifier of the judgement.
-| submission\_id       | ID      | yes       | no        | Identifier of the [ submission](#submissions) judged.
-| judgement\_type\_id  | ID      | yes       | yes       | The [ verdict](#judgement-types) of this judgement.
-| score                | number  | no        | no        | Score for this judgement. Required iff contest:scoreboard_type is `score`.
-| start\_time          | TIME    | yes       | no        | Absolute time when judgement started.
-| start\_contest\_time | RELTIME | yes       | no        | Contest relative time when judgement started.
-| end\_time            | TIME    | yes       | yes       | Absolute time when judgement completed.
-| end\_contest\_time   | RELTIME | yes       | yes       | Contest relative time when judgement completed.
-| max\_run\_time       | number  | no        | yes       | Maximum run time in seconds for any test case. Should be an integer multiple of `0.001`.
+| Name                 | Type      | Description
+| :------------------- | :-------- | :----------
+| id                   | ID        | Identifier of the judgement.
+| submission\_id       | ID        | Identifier of the [ submission](#submissions) judged.
+| judgement\_type\_id  | ID ?      | The [ verdict](#judgement-types) of this judgement.
+| score                | number    | Score for this judgement. Required iff contest:scoreboard_type is `score`.
+| start\_time          | TIME      | Absolute time when judgement started.
+| start\_contest\_time | RELTIME   | Contest relative time when judgement started.
+| end\_time            | TIME ?    | Absolute time when judgement completed.
+| end\_contest\_time   | RELTIME ? | Contest relative time when judgement completed.
```

Comment:
Should these be required iff `judgement_type_id` is present?

Reply:
+1. I think existence of those three should be tied together.

Reply:
Fixed.